### ISSUE TITLE
SpatRasterDataset to list?

### ISSUE THREAD

**Issue by bocinsky** [rspatial/terra#723]
Hi Robert,

I'm really loving working with *terra*, and am really appreciating the speed and elegance of operating on SpatRasterDatasets. However, I'd like to convert a SpatRasterDataset to a list of SpatRaster objects, and I can't figure out for the life of me how to do it. Can you provide any guidance? Thanks!

**Comment by brownag** [comment]
Offhand I don't know if there is a better way, but you can convert a SpatRasterDataset to multilayer SpatRaster, which can then be converted to list. Something like:

```r
library(terra)
x <- system.file("ex", "elev.tif", package="terra")
y <- sds(rast(x), rast(x))
as.list(rast(y))
```

**Comment by rhijmans** [comment]
You can use `as.list` 

```
library(terra)
r <- system.file("ex", "elev.tif", package="terra") |> rast()
x <- c(x,x,x)
names(x) <- letters[1:3]
y <- sds(x[[1:2]],  x[[2:3]])

as.list(y)

#[[1]]
#class       : SpatRaster 
#dimensions  : 90, 95, 2  (nrow, ncol, nlyr)
#resolution  : 0.008333333, 0.008333333  (x, y)
#extent      : 5.741667, 6.533333, 49.44167, 50.19167  (xmin, xmax, ymin, ymax)
#coord. ref. : lon/lat WGS 84 (EPSG:4326) 
#sources     : elev.tif  
#              elev.tif  
#names       :   a,   b 
#min values  : 141, 141 
#max values  : 547, 547 
#
#[[2]]
#class       : SpatRaster 
#dimensions  : 90, 95, 2  (nrow, ncol, nlyr)
#resolution  : 0.008333333, 0.008333333  (x, y)
#extent      : 5.741667, 6.533333, 49.44167, 50.19167  (xmin, xmax, ymin, ymax)
#coord. ref. : lon/lat WGS 84 (EPSG:4326) 
#sources     : elev.tif  
#              elev.tif  
#names       :   b,   c 
#min values  : 141, 141 
#max values  : 547, 547 
```

**Comment by kadyb** [comment]
Note that `as.list()` method for _SpatRasterDataset_ is currently available in the development version (not 1.5-34 which is on CRAN).

In Robert's code `x <- c(x,x,x)` should be `x <- c(r,r,r)`.

**Comment by bocinsky** [comment]
This is perfect Robert. My apologies for not trying it with the dev version.
